src/services/DocumentSelectionRequestProcessor.ts

```diff
@@ -99,6 +104,20 @@ export class DocumentSelectionRequestProcessor {
   		throw new AppError(HttpCodesEnum.BAD_REQUEST, "Missing details in SESSION or PERSON IDENTITY tables");
   	}
 
+  	// Reject the request when session store does not contain email, familyName or GivenName fields
+  	if (!this.validationHelper.checkIfValidString([personDetails.emailAddress])) {
+  		this.logger.error("Missing emailAddress in the PERSON IDENTITY table", {
+  			messageCode: MessageCodes.MISSING_PERSON_EMAIL_ADDRESS,
+  		});
+  		return new Response(HttpCodesEnum.BAD_REQUEST, "Missing emailAddress in the PERSON IDENTITY table");
```

Comment:
Is bad request the best HTTP code for this scenario? My understanding is that it's not a problem with the data in the request, but with the data that we have saved - would something like a server error be more accurate?

Reply:
I was of the same opinion, but i dont know what would the FE do with the error response and respond back to IPV core so that they know the reason for rejecting the request. Hence thought to keep it same as we are doing with Address fields check. @ViveakV your thoughts on this?

Reply:
By "Address fields check" do you mean post office address? Or something else?

Reply:
Yes PO address check. I discussed the same with @ViveakV and decided to change the response to 500 and we can discuss separately how the FE responds.

Reply:
Again, had made these changes and pushed it on thur, but somehow cant see them here. I have pushed them now. Kindly review. Thanks.